src/main/java/cloud/eppo/Configuration.java

```diff
@@ -0,0 +1,203 @@
+  }
+
+  public FlagConfig getFlag(String flagKey) {
+    String flagKeyForLookup = flagKey;
+    if (isConfigObfuscated()) {
+      flagKeyForLookup = getMD5Hex(flagKey);
+    }
+
+    if (flags == null) {
+      log.warn("Request for flag {} before flags have been loaded", flagKey);
+      return null;
+    } else if (flags.isEmpty()) {
+      log.warn("Request for flag {} with empty flags", flagKey);
+    }
+    return flags.get(flagKeyForLookup);
+  }
+
+  public String banditKeyForVariation(String flagKey, String variationValue) {
+    // Note: In practice this double loop should be quite quick as the number of bandits and bandit
+    // variations will be small. Should this ever change, we can optimize things.
+    for (Map.Entry<String, BanditReference> banditEntry : banditReferences.entrySet()) {
+      BanditReference banditReference = banditEntry.getValue();
+      for (BanditFlagVariation banditFlagVariation : banditReference.getFlagVariations()) {
+        if (banditFlagVariation.getFlagKey().equals(flagKey)
+            && banditFlagVariation.getVariationValue().equals(variationValue)) {
+          return banditEntry.getKey();
+        }
+      }
+    }
+    return null;
+  }
+
+  public BanditParameters getBanditParameters(String banditKey) {
+    return bandits.get(banditKey);
+  }
+
+  public boolean isConfigObfuscated() {
+    return isConfigObfuscated;
+  }
+
+  /**
+   * Builder to create the immutable config object.
+   *
+   * @see cloud.eppo.Configuration for usage.
+   */
+  public static class Builder {
+    private static final ObjectMapper mapper =
+        new ObjectMapper().registerModule(EppoModule.eppoModule());
+
+    private final boolean isConfigObfuscated;
+    private final Map<String, FlagConfig> flags;
+    private Map<String, BanditReference> banditReferences;
+    private Map<String, BanditParameters> bandits = Collections.emptyMap();
+
+    public static Configuration emptyConfig() {
+      return new Configuration(
+          Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap(), false);
+    }
+
+    public Builder(String flagJson) {
+      this(flagJson, false);
+    }
+
+    public Builder(String flagJson, boolean isConfigObfuscated) {
+      this.isConfigObfuscated = isConfigObfuscated;
+
+      if (flagJson == null || flagJson.isEmpty()) {
+        throw new RuntimeException(
+            "Null or empty configuration string. Call `Configuration.Builder.Empty()` instead");
+      }
+
+      // Build the flags config from the json string.
+      FlagConfigResponse config;
+      try {
+        config = mapper.readValue(flagJson, FlagConfigResponse.class);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
+      }
+
+      if (config == null || config.getFlags() == null) {
+        log.warn("Flags missing in ufc response");
+        flags = Collections.emptyMap();
+      } else {
+        flags = Collections.unmodifiableMap(config.getFlags());
```

Comment:
minor: I suggest that we store the whole config/bandit responses "as is". There's a couple of other fields that we'll need in the future. e.g., evaluation reasons need environment name and config creation timestamp

In the future, this will also make it easier serializing `Configuration` to a JSON if we want to pass it on to another client (e.g., frontend initialized from backend response)

Reply:
✔️ capturing the JSON string where parsing is successful